src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdServerConfiguration.cs

```diff
@@ -205,5 +222,12 @@ private async Task<OpenIdServerSettings> GetServerSettingsAsync()
 
             return settings;
         }
+
+        private bool IsUseReferenceTokens(OpenIdServerSettings.TokenFormat tokenFormat, bool useReferenceTokens)
```

Comment:
This would be best implemented as a validation check in `OpenIdServerService.ValidateSettingsAsync()`. Something like:

```csharp
if (settings.UseReferenceTokens && settings.AccessTokenFormat == TokenFormat.Encrypted)
{
    results.Add(new ValidationResult(S["Reference tokens can only be enabled when using the default token format."], new[]
    {
        nameof(settings.UseReferenceTokens)
    }));
}
```